app/models/amendment.rb

```diff
@@ -53,6 +53,35 @@ def self.amend_claim(claim, claim_attributes, amendment_attributes)
     amendment
   end
 
+  # Undoes a decision  and makes a new Amendment record with notes and details
+  # on what the decison was previously.
+  #
+  # If the operation fails, then the claim’s changes will not be persisted, and
+  # the Amendment record will not be saved.
+  #
+  # Returns an Amendment record.
+  # - If the operation was successful, this will return true for persisted?
+  # - If the operation failed, this will return false for persisted?, and the
+  # amendment’s errors will have been populated with the errors from the
+  # decision (this will usually only happen if the decision cannot be reversed).
+  def self.undo_decision(decision, amendment_params)
+    amendment = Amendment.new(claim: decision.claim, **amendment_params)
+
+    ActiveRecord::Base.transaction do
+      amendment.claim_changes = {
+        decision: [decision.result, "undecided"]
+      }
+
+      decision.update!(result: "undecided", undone: true)
```

Comment:
Do we want to update the result old decision to `undecided`? I feel like it goes against having the `undone` flag. Now the previous decisions will be both undecided and undone. Should we preserve the old decision result and just mark it undone?

Reply:
Yeah, I did wonder that. I think you're right. Undone is undone, then we can also keep the history a bit better too.